pkg/jsonSchemaValidator/extensions/customKeyRule89.go

```diff
@@ -0,0 +1,81 @@
+// This file defines a custom key to implement the logic for the rule:
+// https://hub.datree.io/built-in-rules/ensure-hostpath-mounts-readonly
+
+package jsonSchemaValidator
+
+import (
+	"fmt"
+
+	"github.com/itchyny/gojq"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+type CustomKeyRule89Compiler struct{}
+
+type CustomKeyRule89Schema map[string]interface{}
+
+var CustomKeyRule89 = jsonschema.MustCompileString("customKeyRule89.json", `{
+	"properties" : {
+		"customKeyRule89": {
+			"type": "string"
+		}
+	}
+}`)
+
+func (CustomKeyRule89Compiler) Compile(ctx jsonschema.CompilerContext, m map[string]interface{}) (jsonschema.ExtSchema, error) {
+	if customKeyRule89, ok := m["customKeyRule89"]; ok {
+		customKeyRule89Str, validStr := customKeyRule89.(map[string]interface{})
+		if !validStr {
+			return nil, fmt.Errorf("customKeyRule89 must be a string")
+		}
+		return CustomKeyRule89Schema(customKeyRule89Str), nil
+	}
+	return nil, nil
+}
+
+type Volume struct {
+	Name string `json:"name"`
+}
+
+func (s CustomKeyRule89Schema) Validate(ctx jsonschema.ValidationContext, dataValue interface{}) error {
+	var hostPathVolumes []string
+
+	query, _ := gojq.Parse(".volumes[] | select(.hostPath != null) | .name")
```

Comment:
here is my suggestion for a refactor:

1. extract method: getHostPathVolumes(dataValue interface{}) []string { ... }
2. extract method: getVolumeMountNames(dataValue interface{}) []string { ... }
3. compare the 2 arrays in the main flow

why? because those are complex and hard-to-read functions that are better of being pure :)